### ISSUE TITLE
Ubuntu 18.04 installation issue

### ISSUE THREAD

**Issue by benjaminirving** [ibme-qubic/quantiphyse#5]
Hi Martin, 

When I try and install on Ubuntu 18.04 with python 2 I get:

```
    -- Could NOT find LibXml2 (missing: LIBXML2_LIBRARY LIBXML2_INCLUDE_DIR) (Required is at least version "2.6.32")
    -- Could NOT find LibXslt (missing: LIBXSLT_LIBRARIES LIBXSLT_INCLUDE_DIR) (Required is at least version "1.1.19")
    Qt QTGUI library not found.
    Qt QTXML library not found.
    Qt QTCORE library not found.
    CMake Error at ApiExtractor/CMakeLists.txt:82 (qt4_add_resources):
      Unknown CMake command "qt4_add_resources".
    
    
    -- Configuring incomplete, errors occurred!
    See also "/tmp/pip-build-JbE5qN/PySide/pyside_build/py2.7-qt5.9.5-64bit-release/shiboken/CMakeFiles/CMakeOutput.log".
    error: Error configuring shiboken
    
    ----------------------------------------
Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-JbE5qN/PySide/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-IUm7bf-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-JbE5qN/PySide/

```
Installing python-pyside and shiboken packages from the Ubuntu package manager doesn't help. 

When I try with Python 3, I get:

```
Collecting quantiphyse
  Using cached https://files.pythonhosted.org/packages/a1/a7/53e97afab67628f860f014a6f0a06cb95237d8b99899f980953a85e9263d/quantiphyse-0.7.7.tar.gz
Requirement already satisfied: Cython in /usr/lib/python3/dist-packages (from quantiphyse) (0.26.1)
Requirement already satisfied: Pillow in /usr/lib/python3/dist-packages (from quantiphyse) (5.1.0)
Collecting PySide (from quantiphyse)
  Using cached https://files.pythonhosted.org/packages/36/ac/ca31db6f2225844d37a41b10615c3d371587677efd074db29855e7035de6/PySide-1.2.4.tar.gz
    Complete output from command python setup.py egg_info:
    only these python versions are supported: [(2, 6), (2, 7), (3, 2), (3, 3), (3, 4)]
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-0t7u4dw3/PySide/
You are using pip version 18.1, however version 19.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

Seems like the biggest issue is PySide. Be great if PySide2 could be supported. PySide2 is released but might have to pip install from a commit of the master branch of pyQTgraph.

**Comment by mcraig-ibme** [comment]
Hi, I will try to get a solid set of install instructions for Ubuntu 16 and 18 this week - it isn't straightforward on either.

Pyside2 is definitely the future - but ideally would like to avoid depending on an unreleased pyqtgraph.

**Comment by benjaminirving** [comment]
Thanks @mcraig-ibme. Great work by the way. 

In my opinion this is probably the biggest blocker to adoption. People want to be able to install easily and on any device. If you test a commit on pyqtgraph and link to that commit then I think it will lower the barrier to entry much further. But just my opinion :)

**Comment by mcraig-ibme** [comment]
I've just run through an installation on Ubuntu 18.04 - the only change from the Ubuntu 16.04 instructions was to use ``sudo apt install python-pip`` to get pip rather than easy_install. I didn't get the error above on building PySide. This was using the default python 2.7

From the looks of it the error would be fixed by adding ``sudo apt install libxml2-dev libxslt-dev`` but this wasn't needed for my build.

**Comment by mcraig-ibme** [comment]
I've updated the installation instructions to include Ubuntu 18.04 and the recipe which worked for me. Let me know if this still fails for you.

I've just tried the latest pyqtgraph with pyside2 and the examples do not work very well. Doing minimal changes in quantiphyse to make it run with pyside2 also show numerous issues with the pyqtgraph components. I'll open a bug specifically for this migration, but there seems there's some way to go before we can move the whole application across.
